### PR TITLE
Update integration tests to use JUnit 5

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

**Review by wburns — APPROVED** [review]
LGTM, waiting on CI

**Comment by ryanemerson** [comment]
Hmm it seems `RestCacheManagerResource` is not executed on Infinispan `main`. I'll look into the failures and fix this PR plus issue one for Infinispan.

**Comment by wburns** [comment]
This https://ci.infinispan.org/job/Infinispan%20Quarkus/job/PR-129/6/testReport/org.infinispan.server.functional.rest/RestServerResource/testMemory/ looks related?

File: integration-tests/server/src/test/java/org/infinispan/quarkus/server/NativeClusteredIT.java [hunk 1]
```diff
@@ -7,27 +7,26 @@
...
-@Suite.SuiteClasses({
+@Suite
+@SelectClasses({
       HotRodCacheOperations.class,
       HotRodCounterOperations.class,
       HotRodMultiMapOperations.class,
       HotRodTransactionalCacheOperations.class,
       RestOperations.class,
       RestRouter.class,
-      RestServerResource.class
```
Comment (ryanemerson) [inline integration-tests/server/src/test/java/org/infinispan/quarkus/server/NativeClusteredIT.java:30]: I've disabled this as it's not enabled on `main` either. To enable this test it requires various substitutions to the ServerResource.

**Pull request merged**